pkg/instancetype/apply/vmi_test.go

```diff
@@ -0,0 +1,1364 @@
+package apply_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
+
+	"kubevirt.io/client-go/api"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+
+	v1 "kubevirt.io/api/core/v1"
+	instancetypev1beta1 "kubevirt.io/api/instancetype/v1beta1"
+
+	instancetypeapply "kubevirt.io/kubevirt/pkg/instancetype/apply"
+)
+
+var _ = Describe("Apply to vmi", func() {
+	var (
+		vmi              *v1.VirtualMachineInstance
+		instancetypeSpec *instancetypev1beta1.VirtualMachineInstancetypeSpec
+		preferenceSpec   *instancetypev1beta1.VirtualMachinePreferenceSpec
+		field            *k8sfield.Path
+	)
+
+	BeforeEach(func() {
+		vmi = api.NewMinimalVMI("testvmi")
+
+		vmi.Spec = v1.VirtualMachineInstanceSpec{
+			Domain: v1.DomainSpec{},
+		}
+		field = k8sfield.NewPath("spec", "template", "spec")
+	})
+
+	Context("instancetype.spec.NodeSelector", func() {
+		It("should apply to VMI", func() {
+			instancetypeSpec = &instancetypev1beta1.VirtualMachineInstancetypeSpec{
+				NodeSelector: map[string]string{"key": "value"},
+			}
+
+			conflicts := instancetypeapply.NewVMIApplier().ApplyToVMI(field, instancetypeSpec, preferenceSpec, &vmi.Spec, &vmi.ObjectMeta)
```

Comment:
nit - call `NewVMIApplier()` in `BeforeEach` to cut the length of this call down a little?